packages/commonwealth/client/scripts/hooks/useInitChainIfNeeded.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useRef } from 'react';
+import { IApp } from 'state';
+import { initChain } from 'helpers/chain';
+
+/**
+ * Use in cases where chain data is needed but hasn't been fetched yet.
+ * Once chain is intialized, the app.chainAdapterReady.emit('ready') event is emitted.
+ * Components that depend on chain data can listen for this to trigger a re-render.
+ *
+ * Helper function initChain requires app.chain to be selected, otherwise this will do nothing.
+ * So this will run after the chain is selected, but not if the chain is loaded already.
+ */
+export const useInitChainIfNeeded = (app: IApp) => {
+  const hasFetchedRef = useRef(false); // prevent multiple calls to initChain
+
+  useEffect(() => {
+    const chainInit = async () => {
+      if (!hasFetchedRef.current && !app.isAdapterReady) {
+        hasFetchedRef.current = true;
+        await initChain();
+      }
+    };
+
+    chainInit();
+  }, [app?.isAdapterReady]);
+};
```

Comment:
>@mhagel Ian added a new chain init call in https://github.com/hicommonwealth/commonwealth/pull/4138 -- can you update this PR to include it, and then we can merge?

@jnaviask Yeah this react hook is actually using the same call already. This just makes it a bit more reusable, and makes sure the initData hasn't already been called.

I just retested. Should be good to go.

Reply:
@mhagel I meant can you add it to the ViewProposalPage, where Ian had to add similar logic in a previous PR? Then we're ready.